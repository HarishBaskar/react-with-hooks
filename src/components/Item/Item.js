import React from 'react';
import { ReactComponent as Check } from '../../SVG/check.svg';
import style from './Item.module.css';

const Item = ({item, onRemoveItem}) => {
    return(
        <div className={style.item}>
          <span style={{width: '40%'}}>
              <a href={item.url}>{item.title}</a>
          </span>
          <span style={{width: '30%'}}>{item.author}</span>
          <br/>
          <span style={{width: '10%', textAlign: 'center'}}>{item.num_comments}</span>
          <br/>
          <span style={{width: '10%'}}>{item.points}</span>
          <br/>
          <span style={{width: '10%'}}>
              <button className={`${style.button} ${style.buttonsmall}`}
                  onClick={() => onRemoveItem(item)}>Dismiss
              &nbsp;
              <Check height="18px" width="18px" />
              </button>
          </span>
      </div>
    )
  }

export default Item;