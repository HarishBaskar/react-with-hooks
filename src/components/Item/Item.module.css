.item{
    display: flex;
    align-items: center;
    padding-bottom: 5px;
}

.item > span{
    padding: 0 5px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.item > span > a{
    color: inherit;
}

.button {
    background: transparent;
    border: 1px solid #171212;
    padding: 5px;
    cursor: pointer;
  
    transition: all 0.1s ease-in;
  }
  
  .button:hover {
    background: #171212;
    color: #ffffff;
  }
  
  .buttonsmall {
    padding: 5px;
  }

  .button:hover > svg > g {
    fill: #ffffff;
    stroke: #ffffff;
  }

  @media screen and (max-width: 767px) {
    .item {
      font-size: 10px;
      display: flex;
      align-items: center;
      width: auto;
      padding-bottom: 5px;
      }

      .button {
        background: transparent;
        border: 1px solid #171212;
        padding: 3px;
        cursor: pointer;
        transition: all 0.1s ease-in;
      }

      .buttonsmall {
        font-size: 8px;
        padding: 5px;
      }

    .item span{
      width: 50px !important;
    }
  }
  
