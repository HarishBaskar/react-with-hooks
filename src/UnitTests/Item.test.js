import React from 'react';
import Item from '../components/Item/Item';
import renderer from 'react-test-renderer';

describe('Item', () => {
    const item = {
        title: 'React',
        url: 'https://reactjs.org/',
        author: 'Jordan Walke',
        num_comments: 3,
        points: 4,
        objectID: 0,
      };

    const handleRemoveItem = jest.fn();

    let component;

    beforeEach(() => {
        component = renderer.create(<Item item={item} onRemoveItem = {handleRemoveItem} />);
    })
    
    it('renders all properties', () => {
        expect(component.root.findByType('a').props.href).toEqual(
            'https://reactjs.org/'
        );

        expect(component.root.findAllByType('span')[1].children.includes('Jordan Walke')).toEqual(true);
    });

    it('calls onRemoveItem on button click', () => {
        component.root.findByType('button').props.onClick();

        expect(handleRemoveItem).toHaveBeenCalledTimes(1);
        expect(handleRemoveItem).toHaveBeenCalledWith(item);

        expect(component.root.findAllByType('span').length).toEqual(5);
    });

    test('renders snapshot', () => {
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
});