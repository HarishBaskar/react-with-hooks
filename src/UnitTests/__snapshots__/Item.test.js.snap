// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders snapshot 1`] = `
<div
  className="item"
>
  <span
    style={
      Object {
        "width": "40%",
      }
    }
  >
    <a
      href="https://reactjs.org/"
    >
      React
    </a>
  </span>
  <span
    style={
      Object {
        "width": "30%",
      }
    }
  >
    Author: 
    Jordan Walke
  </span>
  <br />
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    Comments: 
    3
  </span>
  <br />
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    Points: 
    4
  </span>
  <br />
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <button
      className="button buttonsmall"
      onClick={[Function]}
    >
      Dismiss Â 
      <svg
        height="18px"
        width="18px"
      >
        check.svg
      </svg>
    </button>
  </span>
</div>
`;
